migrations/inputs_docker/migration.go

```diff
@@ -0,0 +1,188 @@
+package inputs_docker
+
+import (
+	"fmt"
+
+	"github.com/influxdata/toml"
+	"github.com/influxdata/toml/ast"
+
+	"github.com/influxdata/telegraf/migrations"
+)
+
+// Migration function to migrate multiple deprecated Docker options
+func migrate(tbl *ast.Table) ([]byte, string, error) {
+	// Decode the old data structure
+	var plugin map[string]interface{}
+	if err := toml.UnmarshalTable(tbl, &plugin); err != nil {
+		return nil, "", err
+	}
+
+	var applied bool
+
+	// 1. Migrate container_names -> container_name_include
+	if containerNamesValue, found := plugin["container_names"]; found {
+		applied = true
+
+		// Convert to []interface{} for easier handling
+		var containerNames []interface{}
+		switch v := containerNamesValue.(type) {
+		case []interface{}:
+			containerNames = v
+		case []string:
+			for _, name := range v {
+				containerNames = append(containerNames, name)
+			}
+		default:
+			return nil, "", fmt.Errorf("container_names value is not a slice: %T", containerNamesValue)
+		}
+
+		// Check if container_name_include already exists
+		if existingInclude, exists := plugin["container_name_include"]; exists {
+			// Merge the arrays
+			var existing []interface{}
+			switch v := existingInclude.(type) {
+			case []interface{}:
+				existing = v
+			case []string:
+				for _, name := range v {
+					existing = append(existing, name)
+				}
+			default:
+				return nil, "", fmt.Errorf("container_name_include value is not a slice: %T", existingInclude)
+			}
+
+			// Append container_names to existing container_name_include
+			merged := append(existing, containerNames...)
+			plugin["container_name_include"] = merged
+		} else {
+			// Create new container_name_include with container_names values
+			plugin["container_name_include"] = containerNames
+		}
```

Comment:
To be honest I find

```suggestion
			// Append container_names to existing container_name_include
			existing := append(existing, containerNames...)
		}
		plugin["container_name_include"] = existing
```

easier to read but that might be a matter of taste...